### PR TITLE
Implicitly cast distinct inline to index.

### PR DESCRIPTION
**Pull request by cbuttner**

### CODE REVIEW

File: src/compiler/sema_casts.c [hunk 1]
```diff
@@ -611,6 +611,13 @@ bool cast_to_index(SemaContext *context, Expr *index)
 		case TYPE_ENUM:
 			type = type->decl->enums.type_info->type->canonical;
 			goto RETRY;
+		case TYPE_DISTINCT:
```
Comment (lerno) [inline src/compiler/sema_casts.c:614]: Even simpler: use "type_flat_distinct_inline" on the type.

Comment (cbuttner) [reply src/compiler/sema_casts.c]: Better?

Comment (lerno) [reply src/compiler/sema_casts.c]: Yes, thank you!

**Comment by lerno** [comment]
Thank you

**Pull request merged**